parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java

```diff
@@ -169,8 +168,7 @@ void writeField(Object value) {
         writer.setFieldName(name);
         writer.setIndex(schema.getFieldIndex(name));
 
-        fieldWriters[i] = writer;
-        i++;
+        fieldWriters[fieldDescriptor.getIndex()] = writer;
```

Comment:
Yes, fields in the list over which we're iterating are ordered by index. Here's how the list is built:
https://github.com/google/protobuf/blob/master/java/core/src/main/java/com/google/protobuf/Descriptors.java#L826